mlx/random.cpp

```diff
@@ -192,6 +193,62 @@ array normal(
   return samples;
 }
 
+array multivariate_normal(
+    const array& mean,
+    const array& cov,
```

Comment:
Could you add a check that the `mean.ndim() >= 1` and `cov.ndim() >= 2`? O/w it will crash in the shape access.